### PR TITLE
Add FPX PaymentMethod Support

### PR DESCRIPTION
**Pull request by shengwei-stripe**
@remi-stripe @ob-stripe
Add Fpx PaymentMethod Support

cc @stripe/api-libraries

### CODE REVIEW

File: charge.go [hunk 1]
```diff
@@ -344,6 +345,12 @@ type ChargePaymentMethodDetailsEps struct {
 	VerifiedName string `json:"verified_name"`
 }
 
+// ChargePaymentMethodDetailsFpx represents details about the FPX PaymentMethod.
+type ChargePaymentMethodDetailsFpx struct {
+	Bank          string `json:"bank"`
```
Comment (remi-stripe) [inline charge.go]: You're missing `account_holder_type` and it should be a list of constants

File: paymentmethod.go [hunk 2]
```diff
@@ -60,12 +60,19 @@ type PaymentMethodCardParams struct {
 	Token    *string `form:"token"`
 }
 
+// PaymentMethodFpxParams is the set of parameters allowed for the `fpx` hash when creating a
+// PaymentMethod of type fpx.
+type PaymentMethodFpxParams struct {
+	Bank *string `form:"bank"`
```
Comment (remi-stripe) [inline paymentmethod.go]: You're missing `account_holder_type`

File: paymentmethod.go [hunk 3]
```diff
@@ -146,11 +153,17 @@ type PaymentMethodCard struct {
 type PaymentMethodCardPresent struct {
 }
 
+type PaymentMethodFpx struct {
+	Bank          string `json:bank`
```
Comment (remi-stripe) [inline paymentmethod.go]: you're missing `account_holdder_type`

File: paymentmethod.go [hunk 4]
```diff
@@ -146,11 +153,17 @@ type PaymentMethodCard struct {
...
+	TransactionId string `json:"transaction_id"`
+}
+
 // PaymentMethod is the resource representing a PaymentMethod.
 type PaymentMethod struct {
 	BillingDetails *BillingDetails           `json:"billing_details"`
 	Card           *PaymentMethodCard        `json:"card"`
 	CardPresent    *PaymentMethodCardPresent `json:"card_present"`
+	Fpx            *PaymentMethodFpx         `json:"fpx"`
```
Comment (remi-stripe) [inline paymentmethod.go]: This is not alphabetically ordered

File: charge.go (hunk 1 above)
Comment (shengwei-stripe) [reply charge.go]: @remi-stripe  done fixing. thanks for the review.

File: charge.go [hunk 5]
```diff
@@ -344,6 +345,13 @@ type ChargePaymentMethodDetailsEps struct {
 	VerifiedName string `json:"verified_name"`
 }
 
+// ChargePaymentMethodDetailsFpx represents details about the FPX PaymentMethod.
+type ChargePaymentMethodDetailsFpx struct {
+	AccountHolderType string `json:"account_holder_type"`
```
Comment (remi-stripe) [inline charge.go]: This needs to be an "enum" not a string on the resource. You would use the same enum that you'd have on the PaymentMethod one really. You'd do something like this: https://github.com/stripe/stripe-go/blob/master/paymentmethod.go#L5-L12

You'd add a `PaymentMethodFpxAccountHolderType` constant in `payment_metod.go` and use it on the resource (params are still `*string`)

Comment (shengwei-stripe) [reply charge.go]: Done @remi-stripe

Comment (remi-stripe) [reply charge.go]: AFAICT it's still not done here, did you not push your commit? It still says `string`

File: paymentmethod.go [hunk 6]
```diff
@@ -2,6 +2,14 @@ package stripe
 
 import "encoding/json"
 
+// List of values that FPX AccountHolderType can take.
+type AccountHolderType string
```
Comment (remi-stripe) [inline paymentmethod.go]: It shouldn't be named `AccountHolderType` here. It should be `PaymentMethodFPXAccountHolderType` instead

File: paymentmethod.go [hunk 7]
```diff
@@ -146,13 +162,20 @@ type PaymentMethodCard struct {
 type PaymentMethodCardPresent struct {
 }
 
+type PaymentMethodFPX struct {
+	AccountHolderType string `json:"account_holder_type"`
```
Comment (remi-stripe) [inline paymentmethod.go]: Same here, still saying `string`

File: paymentmethod.go (hunk 6 above)
Comment (shengwei-stripe) [reply paymentmethod.go]: Done, thanks.

**Comment by shengwei-stripe** [comment]
@remi-stripe  done, finger crossed. Thanks for the review.

File: paymentmethod.go [hunk 8]
```diff
@@ -2,6 +2,15 @@ package stripe
 
 import "encoding/json"
 
+// PaymentMethodFPXAccountHolderType is a ist of string values that FPX AccountHolderType accepts.
+type PaymentMethodFPXAccountHolderType string
+
+// List of values that PaymentMethodFPXAccountHolderType can take
+const (
+	AccountHolderTypeIndividual PaymentMethodFPXAccountHolderType = "individual"
+	AccountHolderTypeCompany    PaymentMethodFPXAccountHolderType = "company"
```
Comment (remi-stripe) [inline paymentmethod.go]: This needs to be `PaymentMethodFPXAccountHolderTypeIndividual`

Comment (shengwei-stripe) [reply paymentmethod.go]: @remi-stripe  Done. sorry for dragging for so long.

File: paymentmethod.go [hunk 9]
```diff
@@ -2,6 +2,15 @@ package stripe
 
 import "encoding/json"
 
+// PaymentMethodFPXAccountHolderType is a ist of string values that FPX AccountHolderType accepts.
```
[the remark this answers is not in the document]
Comment (shengwei-stripe) [reply paymentmethod.go]: Done thanks @remi-stripe

**Pull request merged**